types/match.go

```diff
@@ -103,6 +120,24 @@ func NewRegexMatcher(name model.LabelName, re *regexp.Regexp) *Matcher {
 	}
 }
 
+// NewRouteMatchers returns a new matcher that compares values against values or regular expressions.
+func NewRouteMatchers(t labels.MatchType, n, v string) *Matcher {
+	m := &Matcher{
+		Type:            t,
+		Name:            string(n),
+		Value:           v,
+		NewRouteMatcher: true,
+	}
+	if t == labels.MatchRegexp || t == labels.MatchNotRegexp {
+		re, err := regexp.Compile("^(?:" + v + ")$")
+		if err != nil {
+			return nil
```

Comment:
If this error happens, it is completely silent to the user. The matcher will just not exist (as you return `nil`). That's a big no-no.

As suggested above, we need to check that the regexp (and also the label name and value) is valid during config parsing. Then the compilation cannot fail here.

However, as also suggested (in my option (3) from above), we should probably not need to touch this code here anyawy.

Reply:
Yep, it is not good to ignore this error and I have moved it to config. But I am still a bit confused about unmarshalyaml and where that function will be called.

Reply:
Resolved this